Structure.Rmd

```diff
@@ -0,0 +1,398 @@
+
+You can use `.libPaths()` to see which libraries are currently active. Here are mine:
+
+```{r, eval = FALSE}
+.libPaths()
+#> [1] "/Users/hadley/R"                                               
+#> [2] "/Library/Frameworks/R.framework/Versions/3.1/Resources/library"
+lapply(.libPaths(), dir)
+#> [[1]]
+#>   [1] "AnnotationDbi"   "ash"             "assertthat"     
+#>   ...      
+#> [163] "xtable"          "yaml"            "zoo"            
+#> 
+#> [[2]]
+#>  [1] "base"         "boot"         "class"        "cluster"     
+#>  [5] "codetools"    "compiler"     "datasets"     "foreign"     
+#>  [9] "graphics"     "grDevices"    "grid"         "KernSmooth"  
+#> [13] "lattice"      "MASS"         "Matrix"       "methods"     
+#> [17] "mgcv"         "nlme"         "nnet"         "parallel"    
+#> [21] "rpart"        "spatial"      "splines"      "stats"       
+#> [25] "stats4"       "survival"     "tcltk"        "tools"       
+#> [29] "translations" "utils"
+```
+
+The first lib path is for the packages I've installed (I've installed a lot!). The second is for so-called "recommended" packages that come with every installation of R.
+
+When you use `library(pkg)` or `require(pkg)` to load a package, R looks through each path in `.libPaths()` to see if a directory called `pkg` exists. If it doesn't exist, you'll get an error message:
+
+```{r, error = TRUE}
+library(blah)
+# or:
+require(blah)
+```
+
+The main difference between `library()` and `require()` is what happens if a package isn't found. While `library()` throws an error, `require()` prints a warning message and returns FALSE. In practice, this distinction isn't important because when building a package you should __NEVER__ use either inside a package. See [package dependencies](#dependencies) for what you should do instead.
+
+When you start learning R, it's easy to get confused between libraries and packages because you use `library()` function to load a _package_. However, the distinction between the two is important and useful. For example, one important application is packrat, which automates the process of managing project-specific libraries. With packrat, when you upgrade a package in one project, it only affects that project, not every project on your computer. This is useful because it allows you to play around with cutting-edge packages without affecting other projects' use of older, more reliable packages. This is also useful when you're both developing and using a package.
+t
+
+## Naming your package {#naming}
```

Comment:
Maybe this should now go into a new chapter? Like "Creating a package" or similar?